src/OpenTelemetry.Extensions.Hosting/README.md

```diff
@@ -29,20 +29,24 @@ services.Configure<ZipkinExporterOptions>(this.Configuration.GetSection("Zipkin"
 
 #### Using Dependency Injection
 
-The following example registers a processor of the type "MyProcessor" which has
-been registered as a singleton with the `IServiceCollection`:
+The following example adds a processor instance
+as a singleton to the `IServiceCollection` which will
+be picked up by the TracerProvider.
 
 ```csharp
-services.AddSingleton<MyProcessor>();
+services.AddSingleton<BaseProcessor<Activity>>(new MyProcessor1());
+services.AddSingleton<BaseProcessor<Activity>>(new MyProcessor2());
 
 services.AddOpenTelemetryTracing((builder) => builder
     .AddAspNetCoreInstrumentation()
-    .AddHttpClientInstrumentation()
-    .AddProcessor<MyProcessor>());
+    .AddHttpClientInstrumentation());
 ```
 
-Similar methods exist for registering instrumentation (`AddInstrumentation<T>`)
-and setting a sampler (`SetSampler<T>`).
+Similarly, sampler can also be set on the `IServiceCollection`.
+
+```csharp
+services.AddSingleton<Sampler>(new TestSampler());
```

Comment:
If there are multiple invocations, who would be responsible for disposing the unused one? e.g.:
```csharp
class FooSampler: IDisposable, Sampler
{
}

services.AddSingleton<Sampler>(new FooSampler("A")); // do we leak this?
services.AddSingleton<Sampler>(new FooSampler("B")); // do we always pick the `last one`? or we want to throw exception if there is more than one?
```

https://github.com/open-telemetry/opentelemetry-dotnet/blob/146dd73ed28ddc94b8335306eb62dff5337d783c/src/OpenTelemetry/Trace/TracerProviderSdk.cs#L310